schema.ts

```diff
@@ -1374,6 +1375,16 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    */
   index?: IndexDefinition;
 
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that can be be interpreted as
+   * text/number representing the version of an item.
+   *
+   * Must be a {@link ValueType.String} property.
```

Comment:
this doesn't match your validation code later. I think we decided to force this as a string for now, so probably update the validation code to match this?

Reply:
My thought was that if an API returns a timestamp as a number we should probably avoid the pack maker needlessly creating a string copy of it. So best to proactively support numeric properties even though the contract is that we'll _return_ all versions as strings.

Reply:
Hmmm I defer to @ekoleda-codaio about consistency vs ease-of-use tradeoff.

Either way, let's just make sure comments match reality :)

Reply:
I putted string only initially and forgot to update the comment after. Wdyt the type? @erickoledadevrel

Reply:
One of the things we got right for two-way sync is that the row returned by the code when syncing in is the same as the row value received by the code when executing an update. Returning one type and receiving another is confusing. If we only want to provide strings to the code for now then I would enforce that developers can only return strings in their schema. We can expand that later if we want, but we'd need to support it on both sides.

Reply:
It makes sense to make the type consistent. But if we only support string type, would it make it more complicated to pack makers? For example if the version refers to a numeric timestamp, instead of simply return the property name, the pack maker will have to make change to schema to convert that field into string or add new field with string type.

Reply:
Discussed with Eric and Patrick, the main concern here is like Eric said, it might be confusing to developers on the mismatch of the data type that sent and returned in DB. We can go with string only for the initial version and still have forward compatibility to expand types if needed.